geode-cq/src/distributedTest/java/org/apache/geode/cache/query/cq/dunit/PartitionedRegionCqQueryDUnitTest.java

```diff
@@ -120,6 +123,60 @@
 
   private static int bridgeServerPort;
 
+  @Test
+  public void testPutAllWithCQLocalDestroy() throws Exception {
+    VM server1 = getVM(0);
+    VM server2 = getVM(1);
+    VM client = getVM(2);
+
+    final String cqName = "testPutAllWithCQLocalDestroy_0";
+    createServer(server1);
+    createServer(server2);
+    final String host = Host.getHost(0).getHostName();
+    final int port = server2.invoke(() -> PartitionedRegionCqQueryDUnitTest.getCacheServerPort());
+    createClient(client, port, host);
+    createCQ(client, cqName, cqs[0]);
+
+    int numObjects = 1000;
+
+    server1.invoke(() -> {
+      Region region = getCache().getRegion(SEPARATOR + "root" + SEPARATOR + regions[0]);
+      Map buffer = new HashMap();
```

Comment:
You should avoid raw types in both product and test code. For example: `Region<String, Portfolio>` or even `Region<String, Object>`. Same with Map.

Reply:
PS: You don't really need `SEPARATOR + "root" + SEPARATOR`. This will also work:
```
getCache().getRegion(regions[0]);
```

Reply:
Actually, for this test, we still need `SEPARATOR + "root" + SEPARATOR`, otherwise `getRegion()` can't find the region.